lib/queueservice.js

```diff
@@ -0,0 +1,85 @@
+import xml2js from 'xml2js';
+import Debug from 'debug';
+import request from 'superagent-promise';
+import denodeify from 'denodeify';
+
+const parseXmlString = denodeify(xml2js.parseString.bind(xml2js));
+
+let debug = Debug('taskcluster-docker-worker:queueService');
+
+export default class TaskQueue {
+  constructor(config) {
+    this.queues;
```

Comment:
So a tip I use to make sure that config errors are cached quickly is to validate the input of variables like `config`.
Clearly this is not performance critical, but it's a lot easier to find mismatches between expected input and actual input, if it's validated.

Other wise, those mismatches/bugs shows up at random places during execution which is less fun. Just a general pattern I would recommend, especially when cross module boundaries.
Random example, see doc comment and assert in [queue reaper](https://github.com/taskcluster/taskcluster-queue/blob/1b591a0d03cab96c8b1c9a6ece353e37d44563d9/queue/reaper.js#L8-L35).
Whether you use assert, should.js, expect.js, joi or JSON schemas I don't care... Personally I'm starting to lean towards should.js or expect.js (expect is worse, but should.js don't work well with `undefined`/`null`, it's just too monkey patchy..).